llvm/include/llvm/ADT/SmallVector.h

```diff
@@ -116,8 +116,7 @@ template <class Size_T> class SmallVectorBase {
 
 template <class T>
 using SmallVectorSizeType =
-    std::conditional_t<sizeof(T) < 4 && sizeof(void *) >= 8, uint64_t,
-                       uint32_t>;
+    std::conditional_t<sizeof(T) == 1, uintptr_t, uint32_t>;
```

Comment:
```suggestion
    std::conditional_t<sizeof(T) == 1, size_t, uint32_t>;
```
Unlikely to differ, but I think that is closer to what is intended...